test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/view/resources/OpenShiftODOConnection.java

```diff
@@ -0,0 +1,143 @@
+ ******************************************************************************/
+package org.jboss.tools.openshift.reddeer.view.resources;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.reddeer.common.exception.RedDeerException;
+import org.eclipse.reddeer.common.wait.TimePeriod;
+import org.eclipse.reddeer.common.wait.WaitWhile;
+import org.eclipse.reddeer.swt.api.TreeItem;
+import org.eclipse.reddeer.swt.condition.ShellIsAvailable;
+import org.eclipse.reddeer.swt.impl.menu.ContextMenuItem;
+import org.eclipse.reddeer.workbench.core.condition.JobIsRunning;
+import org.jboss.tools.openshift.reddeer.dialogs.InputDialog;
+import org.jboss.tools.openshift.reddeer.utils.DatastoreOS3;
+import org.jboss.tools.openshift.reddeer.utils.OpenShiftLabel;
+import org.jboss.tools.openshift.reddeer.view.OpenShiftApplicationExplorerView;
+
+/**
+ * 
+ * OpenShift Application Explorer Connection implemented with RedDeer.
+ * 
+ * @author jkopriva@redhat.com
+ *
+ */
+public class OpenShiftODOConnection extends AbstractOpenShiftApplicationConnection {
+	
+	public OpenShiftODOConnection(TreeItem connectionItem) {
+		super(connectionItem);
+	}
+
+	/**
+	 * Gets an OpenShift project with specified project name.
+	 *  
+	 * @param projectName name of a project, if displayed name is provided,
+	 *  	its displayed name, otherwise its project name
+	 * @return OpenShift project
+	 */
+	public OpenShiftODOProject getProject(String projectName) {
+		activateOpenShiftApplicationExplorerView();
+		item.expand();
+		
+		new WaitWhile(new JobIsRunning(), TimePeriod.VERY_LONG);
+		
+		return new OpenShiftODOProject(treeViewerHandler.getTreeItem(item, projectName));
+	}
+
+	/**
+	 * Gets first project of a OpenShift connection.
+	 *
+	 * @return OpenShift project
+	 */
+	public OpenShiftODOProject getProject() {
+		return getProject(DatastoreOS3.PROJECT1_DISPLAYED_NAME);
+	}
+	
+	/**
+	 * Gets all projects existing on a connection.
+	 * 
+	 * @return list of all projects for a connection.
+	 */
+	public List<OpenShiftODOProject> getAllProjects() {
+		List<OpenShiftODOProject> projects = new ArrayList<OpenShiftODOProject>();
+		activateOpenShiftApplicationExplorerView();
+		item.select();
+		item.expand();
+		for (TreeItem treeItem: item.getItems()) {
+			projects.add(new OpenShiftODOProject(treeItem));
+		}
+		return projects;
+	}
+	
+	/**
+	 * Creates a new OpenShift project for a connection.
+	 * 
+	 * @param projectName project name
+	 * @param displayedName displayed name
+	 * @return OpenShift Project
+	 */
+	public OpenShiftODOProject createNewProject(String projectName) {
+		select();
+		new ContextMenuItem(OpenShiftLabel.ContextMenu.NEW_PROJECT).select();
+		
+		InputDialog dialog = new InputDialog(OpenShiftLabel.Shell.NEW_PROJECT);
+		dialog.setInputText(projectName);
+		dialog.ok();
+		
+		new WaitWhile(new ShellIsAvailable(OpenShiftLabel.Shell.NEW_PROJECT), TimePeriod.LONG);
+		new WaitWhile(new JobIsRunning(), TimePeriod.LONG);
```

Comment:
I see a lot of JobIsRunning usage, but if you got some special job for this purpose I would say it is much better to implement specific job wait condition or pass a matcher into JobIsRunning instantiaon in order to wait for specific job.